packages/components/inputs/search-select-input/src/search-select-input.spec.js

```diff
@@ -0,0 +1,320 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { render, fireEvent, waitFor } from '../../../../../test/test-utils';
+import SearchSelectInput from './search-select-input';
+
+const fruits = [
+  { value: 'banana', label: 'Banana' },
+  { value: 'mango', label: 'Mango' },
+  { value: 'raspberry', label: 'Raspberry' },
+  { value: 'lichi', label: 'Lichi' },
+];
+// We use this component to simulate the whole flow of
+// changing a value and formatting on blur.
+class TestSearchSelectComponent extends React.Component {
+  static displayName = 'TestSearchSelectComponent';
+  static propTypes = {
+    id: PropTypes.string,
+    loadOptions: PropTypes.func.isRequired,
+    value: (props, ...rest) =>
+      props.isMulti
+        ? PropTypes.arrayOf(PropTypes.object).isRequired(props, ...rest)
+        : PropTypes.object(props, ...rest),
+    onChange: PropTypes.func,
+  };
+  static defaultProps = {
+    id: 'some-id',
+    name: 'some-name',
+    value: { value: 'banana', label: 'Banana' },
+    loadOptions: (input) =>
+      Promise.resolve(
+        fruits.filter(
+          (fruit) => fruit.label.toLowerCase() === input.toLowerCase()
+        )
+      ),
+  };
+
+  state = {
+    value: this.props.value,
+  };
+
+  handleChange = (event) => {
+    event.persist();
+    this.setState({
+      value: event.target.value,
+    });
+    if (this.props.onChange) {
+      this.props.onChange(event);
+    }
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <label htmlFor={this.props.id}>Fruit</label>
+        <SearchSelectInput
+          {...this.props}
+          onChange={this.handleChange}
+          value={this.state.value}
+          loadOptions={this.props.loadOptions}
+        />
+      </React.Fragment>
+    );
+  }
+}
+
+const renderInput = (props, options) =>
+  render(<TestSearchSelectComponent {...props} />, options);
+
+it('should forward data-attributes', () => {
+  const { container } = renderInput({ 'data-foo': 'bar' });
+  // here we have to use container.querySelector because the data attributes are attached
+  // to the wrapper div, not to the input itself.
+  expect(container.querySelector('[data-foo="bar"]')).toBeInTheDocument();
+});
+
+it('should have focus automatically when isAutofocussed is passed', () => {
+  const { getByLabelText } = renderInput({ isAutofocussed: true });
+  expect(getByLabelText('Fruit')).toHaveFocus();
+});
+
+it('should call onFocus when the input is focused', () => {
+  const onFocus = jest.fn();
+  const { getByLabelText } = renderInput({ onFocus });
```

Comment:
Can't we use screen for this? I recall doing some refactorings to use it in UIKit a while back. The only cause when not to use screen is when you have custom queries not baked into the render.

Reply:
Updated the test to use `screen`